huaweicloud/services/acceptance/dataarts/resource_huaweicloud_dataarts_studio_factory_script_test.go

```diff
@@ -0,0 +1,168 @@
+	rName := acceptance.RandomAccResourceName()
+
+	rc := acceptance.InitResourceCheck(
+		resourceName,
+		&obj,
+		getScriptResourceFunc,
+	)
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck: func() {
+			acceptance.TestAccPreCheck(t)
+			acceptance.TestAccPreCheckDataArtsWorkSpaceID(t)
+			acceptance.TestAccPreCheckDataArtsConnectionName(t)
+		},
+		ProviderFactories: acceptance.TestAccProviderFactories,
+		CheckDestroy:      rc.CheckResourceDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccScript_basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(resourceName, "name", rName),
+					resource.TestCheckResourceAttr(resourceName, "type", "DLISQL"),
+					resource.TestCheckResourceAttr(resourceName, "workspace_id",
+						acceptance.HW_DATAARTS_WORKSPACE_ID),
+					resource.TestCheckResourceAttr(resourceName, "content", "#test"),
+					resource.TestCheckResourceAttr(resourceName, "connection_name",
+						acceptance.HW_DATAARTS_CONNECTION_NAME),
+					resource.TestCheckResourceAttr(resourceName, "directory", "/basic"),
+					resource.TestCheckResourceAttr(resourceName, "queue_name", "default"),
+					resource.TestCheckResourceAttr(resourceName, "description", "test"),
+					resource.TestCheckResourceAttr(resourceName, "configuration.%", "1"),
+					resource.TestCheckResourceAttrSet(resourceName, "created_by"),
+				),
+			},
+			{
+				Config: testAccScript_update(rName),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(resourceName, "name", rName),
+					resource.TestCheckResourceAttr(resourceName, "type", "DLISQL"),
+					resource.TestCheckResourceAttr(resourceName, "workspace_id",
+						acceptance.HW_DATAARTS_WORKSPACE_ID),
+					resource.TestCheckResourceAttr(resourceName, "content", "#test_update"),
+					resource.TestCheckResourceAttr(resourceName, "connection_name",
+						acceptance.HW_DATAARTS_CONNECTION_NAME),
+					resource.TestCheckResourceAttr(resourceName, "directory", "/update"),
+					resource.TestCheckResourceAttr(resourceName, "queue_name", "update"),
+					resource.TestCheckResourceAttr(resourceName, "description", "test_update"),
+					resource.TestCheckResourceAttr(resourceName, "configuration.%", "2"),
+					resource.TestCheckResourceAttrSet(resourceName, "created_by"),
+				),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateIdFunc:       testAccResourceScriptImportStateIDFunc(resourceName),
+				ImportStateVerifyIgnore: []string{"approvers", "target_status"},
+			},
+		},
+	})
+}
+
+func testAccScript_basic(name string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_dataarts_studio_factory_script" "test" {
+  workspace_id    = "%s"
+  name            = "%s"
+  type            = "DLISQL"
+  content         = "#test"
+  connection_name = "%s"
```

Comment:
can we use dataarts_studio_data_connection resource here?

Reply:
of cource.
but the resource dataarts_studio_data_connection will delete failed. So in the ut, it still use variable.